src/app/http-interceptors/http-error-interceptor.ts

```diff
@@ -8,32 +15,37 @@ import {catchError, retry} from 'rxjs/operators';
 @Injectable()
 export class HttpErrorInterceptor implements HttpInterceptor {
   MAX_RETRIES = 5;
+  UNREACHABLE_SERVER = 0;
+  CONNECTION_ERROR_CODES = [
+    this.UNREACHABLE_SERVER,
```

Comment:
What is the difference between 0 and 503 here?

Reply:
0 is when the service is totally down and there's even no response returned. This is the status code I'm getting when I test it locally and the service is not running at all. The 503 Service Unavailable status is being returned by the service when it is not ready to handle the request yet.

Reply:
Normally 0 return code means that everything went well, so it is a bit weird to me.

Reply:
yeah but that's for bash scripts executions

>Something could go wrong on the client-side such as a network error that prevents the request from completing successfully or an exception thrown in an RxJS operator. These errors have status set to 0 and the error property contains a ProgressEvent object, whose type might provide further information.

https://angular.io/guide/http#typed-response